pypeit/core/coadd2d.py

```diff
@@ -788,19 +792,28 @@ def reduce(self, psuedo_dict, show=None, show_peaks=None):
 
         # Make changes to parset specific to 2d coadds
         parcopy = copy.deepcopy(self.par)
-        parcopy['scienceimage']['trace_npoly'] = 3        # Low order traces since we are rectified
+        parcopy['scienceimage']['findobj']['trace_npoly'] = 3        # Low order traces since we are rectified
         #parcopy['scienceimage']['find_extrap_npoly'] = 1  # Use low order for trace extrapolation
-        redux = reduce.instantiate_me(sciImage, self.spectrograph, psuedo_dict['tslits_dict'], parcopy, psuedo_dict['tilts'],
-                                      ir_redux=self.ir_redux, objtype = 'science_coadd2d', det=self.det, binning=self.binning)
+        # Instantiate Calibrations class
+        caliBrate = calibrations.MultiSlitCalibrations(None, parcopy['calibrations'], self.spectrograph)
+        caliBrate.tslits_dict = psuedo_dict['tslits_dict']
+        caliBrate.tilts_dict = dict(tilts=psuedo_dict['tilts'])
+        caliBrate.mswave = psuedo_dict['waveimg']
+        #
+        # redux = reduce.instantiate_me(sciImage, self.spectrograph, psuedo_dict['tslits_dict'], parcopy, psuedo_dict['tilts'],
+        redux=reduce.instantiate_me(sciImage, self.spectrograph, parcopy, caliBrate,
+                                    ir_redux=self.ir_redux, objtype='science_coadd2d',
+                                    det=self.det, binning=self.binning, show=show)
 
         if show:
             redux.show('image', image=psuedo_dict['imgminsky']*(sciImage.mask == 0), chname = 'imgminsky', slits=True, clear=True)
         # Object finding
-        sobjs_obj, nobj, skymask_init = redux.find_objects(sciImage.image, ir_redux=self.ir_redux, show_peaks=show_peaks, show=show)
+        #sobjs_obj, nobj, skymask_init = redux.find_objects(sciImage.image, show_peaks=show_peaks)
+        sobjs_obj, nobj, skymask_init = redux.find_objects(show_peaks=show_peaks)
```

Comment:
This is incorrect. Your new find objects now does other things than my original find objects. So what looks like a minor change is actually a major change. This demonstrates why you should avoid modifying and refactoring codes that you do not fully understand.

Reply:
resolved.

Reply:
This does not have the right argument list now since you changed find_objects.

Reply:
fixed.